test/test-html.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+const fs = require('fs');
+const chalk = require('chalk');
+
+/**
+ * @typedef {import('../types').Identifier} Identifier
+ * @typedef {import('../types').SimpleSupportStatement} SimpleSupportStatement
+ * @typedef {import('../types').CompatStatement} CompatStatement
+ *
+ * @typedef {{error: (...message: unknown[]) => void}} Logger
+ * @typedef {{name:string,nameStart:number,nameEnd:number,value?:string,valueEnd?:number}} AttributeDescriptor
+ */
+
+/** A regular expression used to match HTML elements. */
+const ELEMENT_REGEXP = String.raw`<([a-zA-Z][^\s/>]*)(?: (.*?))?>(.*?)</\1\s*>`;
+/** A regular expression used to match HTML attributes. */
+const ATTR_REGEXP = String.raw`([^\x00-\x20\x7F-\x9F"'>/=\uFDD0-\uFDEF\uFFFE\uFFFF\u{1FFFE}\u{1FFFF}\u{2FFFE}\u{2FFFF}\u{3FFFE}\u{3FFFF}\u{4FFFE}\u{4FFFF}\u{5FFFE}\u{5FFFF}\u{6FFFE}\u{6FFFF}\u{7FFFE}\u{7FFFF}\u{8FFFE}\u{8FFFF}\u{9FFFE}\u{9FFFF}\u{AFFFE}\u{AFFFF}\u{BFFFE}\u{BFFFF}\u{CFFFE}\u{CFFFF}\u{DFFFE}\u{DFFFF}\u{EFFFE}\u{EFFFF}\u{FFFFE}\u{FFFFF}\u{10FFFE}\u{10FFFF}]+)(?: *= *('[^']*'|\\"[^"]*\\"|[^\x09\x0A\x0C\x0D\x20"'=<>\x60]+))?`;
+
+/** Elements that are allowed in all properties. */
+const ALLOWED_GLOBAL_ELEMENTS = [
+  // Force newline
+  'code',
+  'em',
+  'kbd',
+  'strong',
+];
+
+/**
+ * Elements allowed only in specific properties.
+ *
+ * @type {{[property: string]: string[]}}
+ */
+const ALLOWED_PROPERTY_ELEMENTS = {
+  notes: ['a'],
+};
+
+/**
+ * Special attributes that are limited only to specific elements.
+ *
+ * @type {{[element: string]: string[]}}
+ */
+const ALLOWED_ELEMENT_ATTRIBUTES = {
+  a: ['href'],
+};
+
+/**
+ * Calls a defined callback function on each element of an array, and returns an array that contains the results.
+ * @template T, U
+ * @param {T[]} array The array to process.
+ * @param {(value: T, index: number, array: T[]) => U | U[]} callbackfn A function that accepts up to three arguments. The flatMap method calls the callbackfn function one time for each element in the array.
+ * @param {unknown} [thisArg] An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
+ * @return {U[]}
+ */
+function flatMapArray(array, callbackfn, thisArg = this) {
+  /** @type {U[]} */
+  let result = [];
+  for (let i = 0; i < array.length; i++) {
+    result.concat(callbackfn.call(thisArg, array[i], i, array));
+  }
+  return result;
+}
+
+/**
+ * Checks all properties that may contain HTML in the `CompatStatement`.
+ *
+ * @param {CompatStatement} compat The browser compatibility statement.
+ * @param {string} path The path of the feature.
+ * @param {Logger} logger The logger.
+ */
+function checkCompatStatement(compat, path, logger) {
+  let hasErrors = false;
+  if (compat.description) {
+    lintHTMLString(
+      compat.description,
+      'description',
+      `${path}.__compat.description`,
```

Comment:
It looks like `description` is already being mentioned in the error text, and I feel specifying `__compat` is redundant.  We could simplify this as follows:
```suggestion
      path,
```

Example output with the current code: `api.CharacterData.ChildNode.__compat.description - Element <a> is not allowed in property 'description'.`